webapp/views.py

```diff
@@ -511,6 +487,53 @@ def advantage_view():
     )
 
 
+def make_renewal(advantage, contract_info):
+    """Return the renewal as present in the given info, or None."""
+    renewals = contract_info.get("renewals")
+    if not renewals:
+        return None
+
+    renewal = renewals[0]
+
+    # If the renewal is processing, we need to find out
+    # whether payment failed and requires user action,
+    # which is information only available in the fuller
+    # renewal object get_renewal gives us.
+    if renewal["status"] == "processing":
+        renewal = advantage.get_renewal(renewal["id"])
+
+    renewal["renewable"] = False
+
+    # Only actionable renewals are renewable.
+    if not renewal["actionable"]:
+        return renewal
+
+    # The renewal is renewable only during its time window.
+    start = dateutil.parser.parse(renewal["start"])
+    end = dateutil.parser.parse(renewal["end"])
+    if not (start <= datetime.now(timezone.utc) <= end):
+        return renewal
+
+    # Pending renewals are renewable.
+    if renewal["status"] == "pending":
+        renewal["renewable"] = True
+        return renewal
+
+    # Renewals not pending or processing are never renewable.
+    if renewal["status"] != "processing":
+        return renewal
+
+    invoices = renewal.get("stripeInvoices")
+    if invoices:
+        invoice = invoices[-1]
+        renewal["renewable"] = (
+            invoice["pi_status"] == "requires_payment_method"
+            and invoice["subscription_status"] == "incomplete_expired"
```

Comment:
As I understand it, `incomplete_expired` on an invoice subscription status means it's unrenewable (or at least requires intervention from someone on the sales team). 

This check is about presenting the "Renew..." button to the user if they've attempted to pay, failed, left the page and come back later to make another attempt - in that case, the `subscription_status` will be `incomplete`, and is renewable.

```suggestion
            and invoice["subscription_status"] != "incomplete_expired"
```